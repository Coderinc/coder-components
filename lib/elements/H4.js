import PropTypes from 'prop-types';
import styled from 'styled-components';

const H4 = styled.h4`
  color: ${({ theme, color }) => (color ? theme[color] : theme.primaryFont)};
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: ${({ light }) => (light ? 'lighter' : 'bold')};
`;

H4.propTypes = {
  light: PropTypes.bool,
};

H4.defaultProps = {
  light: false,
};

export default H4;
