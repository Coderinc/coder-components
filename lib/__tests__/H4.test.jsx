import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import H4 from '../elements/H4';

describe('H4', () => {
  it('renders correctly', () => {
    const component = renderer.create(<H4>Lorem Ipsum</H4>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('light font', () => {
    const component = renderer.create(<H4 light>Lorem Ipsum</H4>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});