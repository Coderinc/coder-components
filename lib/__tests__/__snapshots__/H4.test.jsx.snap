// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H4 light font 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: lighter;
}

<h4
  className="c0"
>
  Lorem Ipsum
</h4>
`;

exports[`H4 renders correctly 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: bold;
}

<h4
  className="c0"
>
  Lorem Ipsum
</h4>
`;
